src/lib/optimizer/chunk_statistics/range_filter.hpp

```diff
@@ -30,12 +30,32 @@ class RangeFilter : public AbstractFilter {
 
   bool can_prune(const AllTypeVariant& value, const PredicateCondition predicate_type) const override {
     const auto t_value = boost::get<T>(value);
+    // Operators work as follows: value_from_table <operator> t_value
+    // e.g. OpGreaterThan: value_from_table > t_value
+    // thus we can exclude chunk if t_value >= _max since then no value from the table can be greater than t_value
     switch (predicate_type) {
+      case PredicateCondition::GreaterThan: {
+        auto & max = _ranges.back().second;
+        return t_value >= max;
+      }
+      case PredicateCondition::GreaterThanEquals: {
+        auto & max = _ranges.back().second;
+        return t_value > max;
+      }
+      case PredicateCondition::LessThan: {
+        auto & min = _ranges.front().first;
+        return t_value <= min;
+      }
+      case PredicateCondition::LessThanEquals: {
+        auto & min = _ranges.front().first;
+        return t_value < min;
+      }
       case PredicateCondition::Equals: {
-        bool prunable = false;
-        for (const auto& min_max_pair : _ranges) {
-          const auto & [ min, max ] = min_max_pair;
-          prunable |= min < t_value && t_value < max;
+        bool prunable = true;
+        for(const auto& bounds : _ranges) {
+          const auto& [ min, max ] = bounds;
+          // prunable becomes false if t_value is within any of the bounds
+          prunable &= !(t_value >= min && t_value <= max);
```

Comment:
I actually feel slightly queasy about this when `T` is `float`. Especially if `max == min` because then we might decide to prune things we shouldn't.
How are float comparisons supposed to be handled with `PredicateCondition::Equals` in general?

Reply:
I think selecting records based on an equal predicate on a float is something no one would actually do. They code is correct. 

> How are float comparisons supposed to be handled with PredicateCondition::Equals in general?

Good question. I think anything apart from real equality wouldn’t be correct. So a fuzzy comparison doesn’t make sense, does it?

Reply:
Not our problem ;)

https://stackoverflow.com/questions/16149966/comparing-2-float-values-in-sqlserver
https://leehblue.com/match-a-float-in-mysql/